registration/include/pcl/registration/impl/default_convergence_criteria.hpp

```diff
@@ -46,17 +46,26 @@
 template <typename Scalar> bool
 pcl::registration::DefaultConvergenceCriteria<Scalar>::hasConverged ()
 {
-  convergence_state_ = CONVERGENCE_CRITERIA_NOT_CONVERGED;
+  if (convergence_state_ != CONVERGENCE_CRITERIA_NOT_CONVERGED)
+  {
+    //If it already converged or failed before, reset.
+    iterations_similar_transforms_ = 0;
+    convergence_state_ = CONVERGENCE_CRITERIA_NOT_CONVERGED;
+  }
+  
+  bool is_similar = false;
 
   PCL_DEBUG ("[pcl::DefaultConvergenceCriteria::hasConverged] Iteration %d out of %d.\n", iterations_, max_iterations_);
   // 1. Number of iterations has reached the maximum user imposed number of iterations
   if (iterations_ >= max_iterations_)
   {
-    if (failure_after_max_iter_)
-      return (false);
-    
-    convergence_state_ = CONVERGENCE_CRITERIA_ITERATIONS;
-    return (true);
+    if (!failure_after_max_iter_)
+    {
+      iterations_similar_transforms_ = 0;
```

Comment:
I think this line, as well as 82, 98, 110 are not needed anymore. In all of these cases we set state to some kind of convergence. Therefore, if `hasConverged()` is called again, the condition on line 49 will be true and the counter will be reset.

Reply:
OK, I just leave these to make sure the counter explicitly be reset. When I remove it, I'll notice you again.

Reply:
@taketwo Done.